connector/src/main/java/org/geysermc/connector/entity/living/monster/EnderDragonEntity.java

```diff
@@ -81,6 +108,113 @@ public void spawnEntity(GeyserSession session) {
     }
+
+    @Override
+    public boolean despawnEntity(GeyserSession session) {
+        partPositionUpdater.cancel(true);
+
+        for (EnderDragonPartEntity part : allParts) {
+            part.despawnEntity(session);
+        }
+        return super.despawnEntity(session);
+    }
+
+    private void updateBoundingBoxes(GeyserSession session) {
+        Vector3f facingDir = Vector3f.createDirectionDeg(0, rotation.getZ());
+        Segment baseSegment = getSegment(5);
+        // Used to angle the head, neck, and tail when the dragon flies up/down
+        float pitch = (float) Math.toRadians(10 * (baseSegment.getY() - getSegment(10).getY()));
+        float pitchXZ = (float) Math.cos(pitch);
+        float pitchY = (float) Math.sin(pitch);
+
+        // Lowers the head when the dragon sits/hovers
+        float headDuck;
+        if (hovering || metadata.getFlags().getFlag(EntityFlag.SITTING)) {
+            headDuck = -1f;
+        } else {
+            headDuck = baseSegment.y - getSegment(0).y;
+        }
+
+        head.setPosition(facingDir.up(pitchY).mul(pitchXZ, 1, -pitchXZ).mul(6.5f).up(headDuck));
+        neck.setPosition(facingDir.up(pitchY).mul(pitchXZ, 1, -pitchXZ).mul(5.5f).up(headDuck));
+        body.setPosition(facingDir.mul(0.5f, 0f, -0.5f));
+
+        rightWing.setPosition(Vector3f.createDirectionDeg(0, 90f - rotation.getZ()).mul(4.5f).up(2f));
+        leftWing.setPosition(rightWing.getPosition().mul(-1, 1, -1)); // Mirror horizontally
+
+        Vector3f tailBase = facingDir.mul(1.5f);
+        for (int i = 0; i < tail.length; i++) {
+            float distance = (i + 1) * 2f;
+            // Curls the tail when the dragon turns
+            Segment targetSegment = getSegment(12 + 2 * i);
+            float angle = rotation.getZ() + targetSegment.yaw - baseSegment.yaw;
+
+            float tailYOffset = targetSegment.y - baseSegment.y - (distance + 1.5f) * pitchY + 1.5f;
+            tail[i].setPosition(Vector3f.createDirectionDeg(0, angle).mul(distance).add(tailBase).mul(-pitchXZ, 1, pitchXZ).up(tailYOffset));
+        }
+        // Send updated positions
+        for (EnderDragonPartEntity part : allParts) {
+             part.moveAbsolute(session, part.getPosition().add(position), Vector3f.ZERO, false, false);
+        }
+    }
+
+    /**
+     * Store the current yaw and y into the circular buffer
+     */
+    private void pushSegment() {
+        latestSegment = (latestSegment + 1) % segmentHistory.length;
+        segmentHistory[latestSegment].yaw = rotation.getZ();
+        segmentHistory[latestSegment].y = position.getY();
+    }
+
+    /**
+     * Gets the previous yaw and y
+     * Used to curl the tail and pitch the head and tail up/down
+     *
+     * @param index Number of ticks in the past
+     *
+     * @return Segment with the yaw and y
```

Comment:
The line between these should be removed to keep with the javadoc style the rest of the repo uses.